### PR TITLE
feat(relay): Limit custom measurements [INGEST-1620]

### PR DESCRIPTION
**Pull request by jjbayer**
In tandem with https://github.com/getsentry/relay/pull/1483, provide a new project config key that defines how many custom measurements an event may contain.

To distinguish between custom and built-in measurements, the latter are explicitly enumerated as part of project config. This means considerable overhead for every project config, but I believe we need the flexibility here (see linked Relay PR).

### CODE REVIEW

File: src/sentry/relay/config/measurements.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "lcp",
+    "stall_count",
+    "stall_longest_time",
+    "stall_percentage",
+    "stall_total_time",
+    "ttfb.requesttime",
+    "ttfb",
+]
+
+MeasurementName = str
```
Comment (asottile-sentry) [inline src/sentry/relay/config/measurements.py:34]: this defines an alias -- perhaps you meant to use `NewType` ?

https://youtu.be/9rRYeunzX8c

**Review by asottile-sentry — COMMENTED** [review]
just questions -- not blocking

File: src/sentry/relay/config/measurements.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+
+class MeasurementsConfig(TypedDict):
```
[the remark this answers is not in the document]
Comment (jjbayer) [reply src/sentry/relay/config/measurements.py:37]: Yes, this is serialized to JSON and sent to redis / Relay via HTTP. That is also the reason for the camel case names.

File: src/sentry/relay/config/measurements.py (hunk 1 above)
Comment (jjbayer) [reply src/sentry/relay/config/measurements.py:34]: Love the concept, but I _think_ what I want in this case is a simple type alias, to document that the strings in `knownMeasurements` are measurement names. Let me know if you consider this bad style and would prefer putting this information in a docstring.

File: src/sentry/relay/config/measurements.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "cls",
+    "fcp",
+    "fid",
+    "fp",
+    "frames_frozen_rate",
+    "frames_frozen",
+    "frames_slow_rate",
+    "frames_slow",
+    "frames_total",
+    "inp",
```
Comment (jjbayer) [inline src/sentry/relay/config/measurements.py:25]: INP was recently added, see https://github.com/getsentry/sentry/pull/39010.

Comment (iker-barriocanal) [reply src/sentry/relay/config/measurements.py:25]: #39010 is not merged yet, but I guess it's fine?

**Review by iker-barriocanal — APPROVED** [review]

File: src/sentry/relay/config/measurements.py (hunk 3 above)
Comment (jjbayer) [reply src/sentry/relay/config/measurements.py:25]: Yep, should do no harm to already have it here.

File: src/sentry/relay/config/measurements.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class MeasurementsConfig(TypedDict):
+    knownMeasurements: Sequence[MeasurementName]
+    maxCustomMeasurements: int
+
+
+def get_measurements_config() -> MeasurementsConfig:
+    return {
+        "knownMeasurements": BUILTIN_MEASUREMENTS,
```
[the remark this answers is not in the document]
Comment (jjbayer) [reply src/sentry/relay/config/measurements.py:45]: Good suggestion, will replace it in both.

File: src/sentry/relay/config/measurements.py (hunk 1 above)
Comment (asottile-sentry) [reply src/sentry/relay/config/measurements.py:35]: I would just use `str` -- aliases imo are only useful when you've got a complicated type you want to shorten

**Review by iker-barriocanal — APPROVED** [review]

**Pull request merged**